components/movementsensor/mpu6050/mpu6050.go

```diff
@@ -0,0 +1,328 @@
+	}
+
+	var address byte
+	if cfg.UseAlternateI2CAddress {
+		address = 0x69
+	} else {
+		address = 0x68
+	}
+	logger.Debugf("Using address %d for MPU6050 sensor", address)
+
+	backgroundContext, cancelFunc := context.WithCancel(ctx)
+	sensor := &mpu6050{
+		bus:               bus,
+		i2cAddress:        address,
+		logger:            logger,
+		backgroundContext: backgroundContext,
+		cancelFunc:        cancelFunc,
+	}
+
+	// To check that we're able to talk to the chip, we should be able to read register 117 and get
+	// back the device's non-alternative address (0x68)
+	defaultAddress, err := sensor.readByte(ctx, 117)
+	if err != nil {
+		return nil, errors.Errorf("can't read from I2C address %d on bus %s of board %s: '%s'",
+			address, cfg.BusID, cfg.BoardName, err.Error())
+	}
+	if defaultAddress != 0x68 {
+		return nil, errors.Errorf("unexpected non-MPU6050 device at address %d: response '%d'",
+			address, defaultAddress)
+	}
+
+	// The chip starts out in standby mode (the Sleep bit in the power management register defaults
+	// to 1). Set it to measurement mode (by turning off the Sleep bit) so we can get data from it.
+	// To do this, we set register 107 to 0.
+	err = sensor.writeByte(ctx, 107, 0)
+	if err != nil {
+		return nil, errors.Errorf("Unable to wake up MPU6050: '%s'", err.Error())
+	}
+
+	// Now, turn on the background goroutine that constantly reads from the chip and stores data in
+	// the object we created.
+	sensor.activeBackgroundWorkers.Add(1)
+	utils.PanicCapturingGo(sensor.pollData)
+
+	return sensor, nil
+}
+
+func (mpu *mpu6050) readByte(ctx context.Context, register byte) (byte, error) {
+	result, err := mpu.readBlock(ctx, register, 1)
+	if err != nil {
+		return 0, err
+	}
+	return result[0], err
+}
+
+func (mpu *mpu6050) readBlock(ctx context.Context, register byte, length uint8) ([]byte, error) {
+	handle, err := mpu.bus.OpenHandle(mpu.i2cAddress)
+	if err != nil {
+		return nil, err
+	}
+	defer func() {
+		err := handle.Close()
+		if err != nil {
+			mpu.logger.Error(err)
+		}
+	}()
+
+	results, err := handle.ReadBlockData(ctx, register, length)
+	return results, err
+}
+
+func (mpu *mpu6050) writeByte(ctx context.Context, register, value byte) error {
+	handle, err := mpu.bus.OpenHandle(mpu.i2cAddress)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		err := handle.Close()
+		if err != nil {
+			mpu.logger.Error(err)
+		}
+	}()
+
+	return handle.WriteByteData(ctx, register, value)
+}
+
+// A helper function: takes 2 bytes and reinterprets them as a big-endian signed integer.
+func toSignedValue(data []byte) int {
+	return int(int16(binary.BigEndian.Uint16(data)))
+}
+
+// Given a value, scales it so that the range of int16s becomes the range of +/- maxValue.
+func setScale(value int, maxValue float64) float64 {
+	return float64(value) * maxValue / (1 << 15)
+}
+
+// A helper function to abstract out shared code: takes 6 bytes and gives back AngularVelocity.
```

Comment:
[q] what are the units of AngularVelocity here? Are the the same as our contract? Even so please add the unit in the comment.

Reply:
_Very_ good catch! There was a comment 4 lines down about being degrees per second, but the MovementSensor interface is radians per second. I'll scale everything out.